test/unit/host_test.rb

```diff
@@ -1714,6 +1714,14 @@ def teardown
     assert_equal "fqdn=#{host.shortname}.yourdomain.net", LookupValue.find(lookup_value.id).match
   end
 
+  test 'updating host domain should validate domain exists' do
+    host = FactoryGirl.create(:host)
+    fake_domain_id = Domain.last.id + Random.rand(1000000)
```

Comment:
I think it would be a bit safer if you took Doman.order(:id).last.id and no need for here I think